source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.md

```diff
@@ -592,6 +674,27 @@ try:
         if connection is perished:
           close connection
           connection = Null
+        if connection is in "pending response" state:
+          tConnectionDrainingStarted = current instant (use a monotonic clock if possible)
+          emit PendingResponseStartedEvent and equivalent log message
+          drain the pending response
+          if error:
```

Comment:
The mix of exception/error paradigms in this pseudocode block is a bit confusing. We should rewrite this new pseudocode to use try/catch like the existing code.